pkg/controller/networkpolicy/clusternetworkpolicy.go

```diff
@@ -349,15 +387,14 @@ func (n *NetworkPolicyController) processClusterNetworkPolicy(cnp *crdv1beta1.Cl
 			if at.ServiceAccount != nil {
 				atg := n.createAppliedToGroup(at.ServiceAccount.Namespace, serviceAccountNameToPodSelector(at.ServiceAccount.Name), nil, nil, nil)
 				appliedToGroups = mergeAppliedToGroups(appliedToGroups, atg)
-				clusterAppliedToAffectedNS = append(clusterAppliedToAffectedNS, at.ServiceAccount.Namespace)
-				atgForNamespace = append(atgForNamespace, atg)
+				atgPerAffectedNS[at.ServiceAccount.Namespace] = atg
+				affectedNSAndLabels[at.ServiceAccount.Namespace] = n.getNamespaceLabels(at.ServiceAccount.Namespace)
```

Comment:
Does it need to fetch its labels? I guess it's for unification, however, the code would easily panic if someone set ServiceAccount in a non-existing Namespace. It may panic even the Namespace exists due to the cache sync delay.

Reply:
Fixed